src/vaults/fuse/FuseERC4626.sol

```diff
@@ -56,6 +56,46 @@ contract FuseERC4626 is ERC4626 {
         require(cToken.mint(underlyingAmount) == 0, "MINT_FAILED");
     }
 
+    function deposit(uint256 amount, address receiver)
+        public
+        override
+        returns (uint256)
+    {
+        require(amount != 0, "ZERO_ASSETS"); // Deposit 0 makes no sense
+        require(amount <= type(uint128).max, "MANY_ASSETS"); // Check for overly large values
```

Comment:
why add this uint128 check? I don't think its necessary

Reply:
on large values the CToken fails with `MINT_EXCHANGE_CALCULATION_FAILED` and there can be issues with wadMul

Reply:
right but its gas intensive to add. In practice token values don't get this large